nssa/src/state.rs

```diff
@@ -2221,6 +2223,1547 @@ pub mod tests {
+        AccountId::from((
+            &amm_program_id,
+            &compute_pool_pda_seed(definition_token_a_id, definition_token_b_id),
+        ))
+    }
+
+    fn compute_pool_pda_seed(
+        definition_token_a_id: AccountId,
+        definition_token_b_id: AccountId,
+    ) -> PdaSeed {
+        use risc0_zkvm::sha::{Impl, Sha256};
+
+        let mut i: usize = 0;
+        let (token_1, token_2) = loop {
+            if definition_token_a_id.value()[i] > definition_token_b_id.value()[i] {
+                let token_1 = definition_token_a_id.clone();
+                let token_2 = definition_token_b_id.clone();
+                break (token_1, token_2);
+            } else if definition_token_a_id.value()[i] < definition_token_b_id.value()[i] {
+                let token_1 = definition_token_b_id.clone();
+                let token_2 = definition_token_a_id.clone();
+                break (token_1, token_2);
+            }
+
+            if i == 32 {
+                panic!("Definitions match");
+            } else {
+                i += 1;
+            }
+        };
+
+        let mut bytes = [0; 64];
+        bytes[0..32].copy_from_slice(&token_1.to_bytes());
+        bytes[32..].copy_from_slice(&token_2.to_bytes());
+
+        PdaSeed::new(
+            Impl::hash_bytes(&bytes)
+                .as_bytes()
+                .try_into()
+                .expect("Hash output must be exactly 32 bytes long"),
+        )
+    }
+
+    fn compute_vault_pda(
+        amm_program_id: ProgramId,
+        pool_id: AccountId,
+        definition_token_id: AccountId,
+    ) -> AccountId {
+        AccountId::from((
+            &amm_program_id,
+            &compute_vault_pda_seed(pool_id, definition_token_id),
+        ))
+    }
+
+    fn compute_vault_pda_seed(pool_id: AccountId, definition_token_id: AccountId) -> PdaSeed {
+        use risc0_zkvm::sha::{Impl, Sha256};
+
+        let mut bytes = [0; 64];
+        bytes[0..32].copy_from_slice(&pool_id.to_bytes());
+        bytes[32..].copy_from_slice(&definition_token_id.to_bytes());
+
+        PdaSeed::new(
+            Impl::hash_bytes(&bytes)
+                .as_bytes()
+                .try_into()
+                .expect("Hash output must be exactly 32 bytes long"),
+        )
+    }
+
+    fn compute_liquidity_token_pda(amm_program_id: ProgramId, pool_id: AccountId) -> AccountId {
+        AccountId::from((&amm_program_id, &compute_liquidity_token_pda_seed(pool_id)))
+    }
+
+    fn compute_liquidity_token_pda_seed(pool_id: AccountId) -> PdaSeed {
+        use risc0_zkvm::sha::{Impl, Sha256};
+
+        let mut bytes = [0; 64];
+        bytes[0..32].copy_from_slice(&pool_id.to_bytes());
+        bytes[32..].copy_from_slice(&[0; 32]);
+
+        PdaSeed::new(
+            Impl::hash_bytes(&bytes)
+                .as_bytes()
+                .try_into()
+                .expect("Hash output must be exactly 32 bytes long"),
+        )
+    }
+const POOL_DEFINITION_DATA_SIZE: usize = 225;
```

Comment:
Ok, what about 
```suggestion
const POOL_DEFINITION_DATA_SIZE: usize = std::mem::size_of<PoolDefinition>();

const _: () = assert!(POOL_DEFINITION_DATA_SIZE == 225);
```

And the same in other places?

So that this number and actual data size won't get out of sync

Reply:
This returns a size of 240. I think this is due to the round up from 225 to account for the next u128.

Does this make sense to use?